marconi/test/Spec/Cli.hs

```diff
@@ -0,0 +1,109 @@
+{-# LANGUAGE OverloadedStrings #-}
+
+module Spec.Cli where
+
+import Control.Monad (forM_)
+import Data.ByteString.Lazy qualified as BSL
+import Data.Text qualified as T
+import Data.Text.Encoding (encodeUtf8)
+import Options.Applicative (ParserInfo, ParserResult (CompletionInvoked, Failure, Success), defaultPrefs,
+                            execParserPure, renderFailure, (<**>))
+import Options.Applicative qualified as OptA
+import Test.Tasty (TestTree, testGroup)
+import Test.Tasty.Golden (goldenVsStringDiff)
+import Test.Tasty.HUnit (Assertion, assertBool, testCase)
+
+import Marconi.CLI (Options, optionsParser)
+
+tests :: TestTree
```

Comment:
Do we actually need to validate the output of the command?

Reply:
some commands are optional and some required. the intent was to test those.

Reply:
I think the correctness of the cli is already "proved" by the fact that you can start the desired indexer with it (e.g without needing to fill in missing values with `undefined` in Main).

The cli is generated from a data type for arguments, and this same arguments type is used to start the indexer, so if both of those are implemented (= you can start indexer by running the exe) then it should be fine already with that and testing cli output is not required. (If there is fear that optparse-applicative misses spots, then the tests would be better of in that library instead)

Reply:
CLI golden tests are removed from both, marconi, and marconi-mamba